SwiftCompilerSources/Sources/CMakeLists.txt

```diff
@@ -9,5 +9,6 @@
 if(SWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING)
   add_subdirectory(ExperimentalRegex)
 endif()
+add_subdirectory(AST)
```

Comment:
Please put SubstitutionMap directly into SIL. 
It's like SILType, which is a wrapper around the AST Type. Except that in C++ we don't have a SIL wrapper around SubstitutionMap, but that doesn't matter for the swift SubstitutionMap.